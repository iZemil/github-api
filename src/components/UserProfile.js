import React, { Component } from 'react';

export default class UserProfile extends Component {

  render() {
    const user = this.props.user.data;

    return (
      user.isFetching ? 
      <div className="user-profile">
        <h2>Профиль пользователя</h2>
        <img src={user.avatar_url} className="user-profile__avatar" alt="avatar" />
        <div className="user-profile__data">
          <p>@{user.login}</p>
          <p>Имя: {user.name}</p>
          <p>Открытых репозиториев: {user.public_repos}</p>
          <a href={user.url}>Ссылка на Github</a>
        </div>
      </div>
       : null
    );
  }
}

