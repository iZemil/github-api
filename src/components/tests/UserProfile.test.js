import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import UserProfile from '../UserProfile';

describe('AboutPage.js', () => {
  const renderedComponent = shallow(
    <Home username={'Alice'} changeUsername={jest.fn()} />
  );

  it('Рендер компонента UserProfile без ошибок', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserProfile />, div);
  })
});