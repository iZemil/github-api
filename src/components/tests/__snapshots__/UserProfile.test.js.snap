// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile.js Рендер компонента UserProfile без ошибок 1`] = `
<div
  className="user-profile"
>
  <h2>
    Профиль пользователя
  </h2>
  <img
    alt="avatar"
    className="user-profile__avatar"
    src="http://avatarka.ru/13"
  />
  <div
    className="user-profile__data"
  >
    <p>
      @
      iZemil
    </p>
    <p>
      Имя: 
      Emil Zaripov
    </p>
    <p>
      Открытых репозиториев: 
      6
    </p>
    <a
      href="https://github.com/"
    >
      Ссылка на Github
    </a>
  </div>
</div>
`;
